portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java

```diff
@@ -1680,10 +1680,14 @@ else if (_PATH_SEPARATOR_FILE_ENTRY.equals(pathArray[0])) {
 			Optional<FileEntry> fileEntryOptional = _resolveFileEntry(
 				httpServletRequest, pathArray);
 
-			return fileEntryOptional.orElseThrow(
+			FileEntry fileEntry = fileEntryOptional.orElseThrow(
 				() -> new NoSuchFileEntryException(
 					"No file entry found for friendly URL " +
 						Arrays.toString(pathArray)));
+
+			_checkExpiredFileEntry(fileEntry, httpServletRequest);
```

Comment:
Why not make the permission checks in the WebServerServlet as well? That way the lower layers can keep using the local service (that makes the useful in more contexts).

Reply:
Because I wanted to mimic the behavior on the  DLAppServiceUtil.getFileEntryByUuidAndGroupId but I ended making the thing too complex.